.circleci/config.yml

```
@@ -0,0 +1,557 @@
+version: 2
+
+defaults: &defaults
+  docker:
+    - image: palantirtechnologies/circle-spark-base:0.2.4
+  resource_class: xlarge
+  environment: &defaults-environment
+    TERM: dumb
+    BUILD_SBT_CACHE: "/home/circleci/build-sbt-cache"
+
+test-defaults: &test-defaults
+  <<: *defaults
+  environment:
+    <<: *defaults-environment
+    CIRCLE_TEST_REPORTS: /tmp/circle-test-reports
+    TEST_RESULTS_FILE: /tmp/test-results/results.json
+
+
+all-branches-and-tags: &all-branches-and-tags
+  filters:
+    # run on all branches and tags
+    tags:
+      only: /.*/
+
+deployable-branches-and-tags: &deployable-branches-and-tags
+  filters:
+    tags:
+      only: /.*/
+    branches:
+      only:
+       - master
+       - jc/pt-diff-on-3.0-base
```

```suggestion
       - jc/pt-diff-in-logical-chunks
```

Also probably want to remove before merging.